pkg/endpoint/bpf.go

```diff
@@ -134,13 +134,24 @@ func (e *Endpoint) writeHeaderfile(prefix string) error {
 	f, err := os.Create(headerPath)
 	if err != nil {
 		return fmt.Errorf("failed to open file %s for writing: %s", headerPath, err)
-
 	}
 	defer f.Close()
 
 	if err = e.writeInformationalComments(f); err != nil {
 		return err
 	}
+
+	// Create symlink with old header filename, to allow downgrade to pre-1.8
+	// Cilium. Can be removed once v1.8 is the oldest supported release.
+	// We don't add the symlink for the host endpoint so that it is not
+	// restored when downgrading to <1.8.
+	if !e.IsHost() {
+		oldHeaderPath := filepath.Join(prefix, common.OldCHeaderFileName)
+		if err := os.Symlink(common.CHeaderFileName, oldHeaderPath); err != nil {
+			e.getLogger().Warn("Failed to create C header file symlink")
+		}
+	}
```

Comment:
What kind of validation have you done to ensure that the implementation will handle the symlink correctly? (Do we think that the standard upgrade test should properly handle this?)

Reply:
I validated this using the existing K8sUpdates test (doing upgrades and downgrades between v1.7 and master). It took several iterations to get this right so I know it is actually tested. I also checked manually the set of files and symlinks present at different points during the test (e.g., after downgrade, after upgrade).